src/libfuncs/felt252.rs

```diff
@@ -253,47 +249,34 @@ pub fn build_binary_operation<'ctx, 'this>(
             let inverse = loop_block.argument(3)?.into();
 
             // First calculate q = rem_(i-1)/rem_i, rounded down
-            let quotient = loop_block
-                .append_operation(arith::divui(prev_remainder, remainder, location))
-                .result(0)?
-                .into();
+            let quotient =
+                loop_block.append_op_result(arith::divui(prev_remainder, remainder, location))?;
             // Then r_(i+1) = r_(i-1) - q * r_i, and inv_(i+1) = inv_(i-1) - q * inv_i
-            let rem_times_quo = loop_block
-                .append_operation(arith::muli(remainder, quotient, location))
-                .result(0)?
-                .into();
-            let inv_times_quo = loop_block
-                .append_operation(arith::muli(inverse, quotient, location))
-                .result(0)?
-                .into();
-            let next_remainder = loop_block
-                .append_operation(arith::subi(prev_remainder, rem_times_quo, location))
-                .result(0)?
-                .into();
-            let next_inverse = loop_block
-                .append_operation(arith::subi(prev_inverse, inv_times_quo, location))
-                .result(0)?
-                .into();
+            let rem_times_quo =
+                loop_block.append_op_result(arith::muli(remainder, quotient, location))?;
+            let inv_times_quo =
+                loop_block.append_op_result(arith::muli(inverse, quotient, location))?;
+            let next_remainder = loop_block.append_op_result(arith::subi(
+                prev_remainder,
+                rem_times_quo,
+                location,
+            ))?;
+            let next_inverse =
+                loop_block.append_op_result(arith::subi(prev_inverse, inv_times_quo, location))?;
 
             // If r_(i+1) is 0, then inv_i is the inverse
-            let zero = loop_block
-                .append_operation(arith::constant(
-                    context,
-                    IntegerAttribute::new(i512, 0).into(),
-                    location,
-                ))
-                .result(0)?
-                .into();
-            let next_remainder_eq_zero = loop_block
-                .append_operation(arith::cmpi(
-                    context,
-                    CmpiPredicate::Eq,
-                    next_remainder,
-                    zero,
-                    location,
-                ))
-                .result(0)?
-                .into();
+            let zero = loop_block.append_op_result(arith::constant(
+                context,
+                IntegerAttribute::new(i512, 0).into(),
+                location,
+            ))?;
```

Comment:
Here I think you can just use the block ext `const_int` method no?

Reply:
If the block ext `const_int` method is used there it throws an error that 4 arguments are expected and 1 is missing, I am not sure if I can add one more and what would it be.

Reply:
Here it should be 

```rust
.const_int(context, location, 0, 512)?
```

I think, 512 being the number of bits for your `i512`, you can check if it works with other examples of the codebase and some refactors being done here for example 

https://github.com/lambdaclass/cairo_native/pull/648

Reply:
thank you! I updated the PR!